diff --git a/node_modules/@aave/contract-helpers/src/commons/BaseService.ts b/node_modules/@aave/contract-helpers/src/commons/BaseService.ts
index 8200988..21c9048 100644
--- a/node_modules/@aave/contract-helpers/src/commons/BaseService.ts
+++ b/node_modules/@aave/contract-helpers/src/commons/BaseService.ts
@@ -64,7 +64,15 @@ export default class BaseService<T extends Contract> {
         value: value ?? DEFAULT_NULL_VALUE_ON_TX,
       };
 
+      console.log("test estimateGasByNetwork")
+      try{
       tx.gasLimit = await estimateGasByNetwork(tx, this.provider, gasSurplus);
+    }catch(e){
+        console.log("[handled] - estimateGasByNetwork failed", e);
+        tx.gasLimit = BigNumber.from(
+            gasLimitRecommendations[action].recommended,
+          );
+        };
 
       if (
         action &&
@@ -77,7 +85,7 @@ export default class BaseService<T extends Contract> {
       }
 
       return tx;
-    };
+    
 
   readonly generateTxPriceEstimation =
     (
